rostime/src/time.cpp

```diff
@@ -167,6 +167,29 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_monotonictime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+#if HAS_CLOCK_GETTIME
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    // what to do if clock_gettime is not available???
+    struct timeval timeofday;
+    gettimeofday(&timeofday,NULL);
+    sec  = timeofday.tv_sec;
+    nsec = timeofday.tv_usec * 1000;
```

Comment:
Thanks...
On a sidenote:
I'm not convinced that it is a good idea to use the `CLOCK_MONOTONIC_RAW` (even if its available) for steady time. IMHO `CLOCK_MONOTONIC` is always what we want here, namely it doesn't jump, but the clock drift is adjusted...